course_access_groups/permissions.py

```diff
@@ -10,13 +10,28 @@
     SessionAuthentication,
     TokenAuthentication,
 )
+from django.contrib.sites.models import Site
 from django.contrib.sites import shortcuts as sites_shortcuts
 from openedx.core.lib.api.authentication import OAuth2Authentication
 from rest_framework.permissions import IsAuthenticated
 from organizations.models import Organization, UserOrganizationMapping
 from rest_framework.permissions import BasePermission
 
 
+def get_current_organization(request):
+    """
+    Return a single organization for the current site.
+
+    :param request:
+    :raise Site.DoesNotExist when the site isn't found.
+    :raise Organization.DoesNotExist when the organization isn't found.
+    :raise Organization.MultipleObjectsReturned when more than one organization is returned.
+    :return Organization.
+    """
+    current_site = sites_shortcuts.get_current_site(request)
+    return Organization.objects.get(sites__in=[current_site])
```

Comment:
What if there are multiple organizations in a site? This will fail

Reply:
Yes @johnbaldwin, so far we have a 1:1 site/org relationship. I need to ensure that we don't have funny security side effects.

Reply:
Right now it looks like the user/api client will get a 500 error if there are multiple orgs map to a given site, which is probably more likely than it may look since

A. edx-platform/edx-organizations allows for multiple orgs to map to a single site
B. From what I understand, the org-site mapping logic is not centralized in one location in the platform/support apps. Perhaps this is something that we can address globally in the unification investigative and design phase

Reply:
@OmarIthawi Would you please create a bug card for this? Since we still have work and testing to do before production, I don't want to hold up your progress by not accepting this PR, because otherwise, I think it looks good, but I'm also pretty worried that if we don't capture this issue for follow up, it will come back to bite us at the worst possible time

Reply:
@johnbaldwin I did create a GitHub issue #29. Thanks for pushing for that. 

> Right now it looks like the user/api client will get a 500 error if there are multiple orgs map to a given site, which is probably more likely than it may look since

Yeah, I guess so. I think we can contain such errors to make them more user-friendly.

Reply:
Thanks @OmarIthawi !